Functions/QualityControl/xASL_qc_CleanOldQC.m

```diff
@@ -0,0 +1,94 @@
+function [x] = xASL_qc_CleanOldQC(x, ScanType, bRemoveCurrentSession)
+%xASL_qc_RemoveOutdatedQC Removes outdated QC parameters, loads QC from JSON and cleans the current session if necessary
+%
+% FORMAT: [x] = xASL_qc_RemoveOutdatedQC(x, ScanType[, bRemoveCurrentSession)
+%
+% INPUT:
+%   x 	                  - structure containing fields with all information required to run this submodule (REQUIRED)
+%   ScanType              - Type of the data - can be 'ASL' or 'func' (REQUIRED)
+%   bRemoveCurrentSession - Removes the current session completely (OPTIONAL, DEFAULT = false)
+%
+% OUTPUT:
+%   x        - same as input
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function removes the old and outdated QC parameters. Specifically parameters for ASL and func which are now given in a subfield by session.
+%              It also loads further sessions from the disc and if necessary, it cleans the current session completely.
+%
+% EXAMPLE: x = xASL_qc_RemoveOutdatedQC(x, 'ASL');
+% __________________________________
+% Copyright (C) 2015-2023 ExploreASL	
+
+if nargin < 2
+	error('Two parameters are required');
+end
+
+if nargin < 3 || isempty(bRemoveCurrentSession)
+	bRemoveCurrentSession = false;
+end
+
+% All possible scantype lists
+ScanTypeList = {'func', 'ASL', 'dwi'};
+
+% Load QC.json and save it to x-struct
+QC_Path = fullfile(x.D.ROOT, x.SUBJECT, ['QC_collection_' x.SUBJECT '.json']);
```

Comment:
Can you add headers to the code?
Something like:

1. Manage potentially old x.Output without sessions (here)
2. Fully remove current session (line 62)
3. Remove old x.Output without sessions (line 73)

but you might want to rename (1) because this is something different from (3) right?
And when we have these headers, it is also easier for me to check if the order is useful (e.g. perhaps if you only remove the current session, you don't have to run through the rest of the code)

Reply:
Good point. The comments and explanation were not really clear. This is now improved in c523326a